modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/ZoomGestureRecognizer.java

```diff
@@ -278,10 +271,11 @@ public void notifyEndTouchEvent(long time) {
                         totalZoomFactor *= zoomFactor;
                         sendZoomEvent(false);
                         distanceReference = currentDistance;
-                        double timePassed = ((double)time - zoomStartTime) / 1000000000;
-                        if (timePassed > 1e-4) {
-                            initialInertiaZoomVelocity = (totalZoomFactor - prevTotalZoomFactor) / timePassed;
-                            zoomStartTime = time;
+                        long nanosPassed = nanos - zoomStartNanos;
+
+                        if (nanosPassed > INITIAL_VELOCITY_THRESHOLD_NANOS) {
+                            initialInertiaZoomVelocity = (totalZoomFactor - prevTotalZoomFactor) / nanosPassed * NANOS_TO_SECONDS;
```

Comment:
`(totalZoomFactor - prevTotalZoomFactor) / (nanosPassed * NANOS_TO_SECONDS);` ?

Reply:
The order is irrelevant, it could as well be `((totalZoomFactor - prevTotalZoomFactor) / nanosPassed) * NANOS_TO_SECONDS` -- I'll let the compiler decide what is more optimal.

Reply:
See other conversation, your suggestion would not be correct as it changes the order of the operations.